aten/src/ATen/native/mkldnn/RNN.cpp

```diff
@@ -553,9 +553,8 @@ void lstm_mkldnn(Tensor& output, Tensor& hy, Tensor& cy,
     int64_t num_layers, double dropout_p, bool train, bool bidirectional, bool batch_first) {
   auto result = mkldnn_impl(input, std::make_tuple(hx[0], hx[1]), params, has_biases,
       ideep::rnn_kind::LSTM, num_layers, dropout_p, train, bidirectional, batch_first);
-  output = result.first;
-  hy = std::get<0>(result.second);
-  cy = std::get<1>(result.second);
+  output = std::move(result.first);
```

Comment:
You can unpack the result using a combination of forward_as_tuple std::tie at the end using forward_as_tuple if you wish to avoid any temporary output. May not be worth the readability hit though.